services/sqs/src/test/java/software/amazon/awssdk/services/sqs/SqsBatchManagerTest.java

```diff
@@ -0,0 +1,345 @@
+        String id1 = "0";
+        String id2 = "1";
+        String errorCode = "400";
+        String errorMessage = "Some error";
+        String responseBody = String.format(
+            "<SendMessageBatchResponse>\n"
+            + "<SendMessageBatchResult>\n"
+            + "    <BatchResultErrorEntry>\n"
+            + "        <Id>%s</Id>\n"
+            + "        <Code>%s</Code>\n"
+            + "        <Message>%s</Message>\n"
+            + "    </BatchResultErrorEntry>\n"
+            + "    <BatchResultErrorEntry>\n"
+            + "        <Id>%s</Id>\n"
+            + "        <Code>%s</Code>\n"
+            + "        <Message>%s</Message>\n"
+            + "    </BatchResultErrorEntry>\n"
+            + "</SendMessageBatchResult>\n"
+            + "</SendMessageBatchResponse>", id1, errorCode, errorMessage, id2, errorCode, errorMessage);
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(200).withBody(responseBody)));
+
+        List<SendMessageRequest> requests = new ArrayList<>();
+        requests.add(createSendMessageRequest(id1));
+        requests.add(createSendMessageRequest(id2));
+
+        List<CompletableFuture<SendMessageResponse>> responses = new ArrayList<>();
+        for (SendMessageRequest request : requests) {
+            responses.add(batchManager.sendMessage(request));
+        }
+
+        SendMessageResponse completedResponse1 = responses.get(0).join();
+        SendMessageResponse completedResponse2 = responses.get(1).join();
+        String expectedHash = getMd5Hash(String.format("%s: %s", errorCode, errorMessage));
+        Assert.assertEquals(expectedHash, completedResponse1.md5OfMessageBody());
+        Assert.assertEquals(expectedHash, completedResponse2.md5OfMessageBody());
+    }
+
+    @Test
+    public void sendMessageBatchFunctionReturnsWithError_completeMessagesExceptionally() {
+        String id1 = "0";
+        String id2 = "1";
+        String responseBody = "<Error>\n"
+                              + "<Code>CustomError</Code>\n"
+                              + "<Message>Foo bar</Message>\n"
+                              + "<RequestId>656c76696e6727732072657175657374</RequestId>\n"
+                              + "<HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>\n"
+                              + "</Error>";
+
+        stubFor(any(anyUrl()).willReturn(aResponse().withStatus(400).withBody(responseBody)));
+
+        List<SendMessageRequest> requests = new ArrayList<>();
+        requests.add(createSendMessageRequest(id1));
+        requests.add(createSendMessageRequest(id2));
+
+        List<CompletableFuture<SendMessageResponse>> responses = new ArrayList<>();
+        for (SendMessageRequest request : requests) {
+            responses.add(batchManager.sendMessage(request));
+        }
+
+        Assert.assertThrows(Exception.class, () -> responses.get(0).join());
+        Assert.assertThrows(Exception.class, () -> responses.get(1).join());
```

Comment:
Let's use assertj `assertThatThrownBy` instead and verify the exact exception class instead of the generic Exception

Example: https://github.com/aws/aws-sdk-java-v2/blob/master/services/rds/src/test/java/software/amazon/awssdk/services/rds/DefaultRdsUtilitiesTest.java#L106

Reply:
Used `assertThatThrownBy` to check for a CompletionException.